big_tests/tests/mod_global_distrib_SUITE.erl

```diff
@@ -230,20 +230,27 @@ init_per_testcase(CaseName, Config)
     muc_helper:load_muc(<<"muc.localhost">>),
     escalus:init_per_testcase(CaseName, Config);
 init_per_testcase(test_pm_with_graceful_reconnection_to_different_server = CN, Config) ->
-    EveSpec = escalus_fresh:freshen_spec(Config, eve),
-    escalus:create_users(Config, [{eve, [{port, 5222} | EveSpec]}]),
-    escalus:init_per_testcase(CN, [{evespec_mim, EveSpec} |Config]);
+    escalus:init_per_testcase(CN, init_user_eve(Config));
 init_per_testcase(test_pm_with_ungraceful_reconnection_to_different_server = CN, Config) ->
-    EveSpec = escalus_fresh:create_fresh_user(Config, eve),
-    EveSpec2 = lists:keystore(port, 1, EveSpec, {port, 5222}),
-    escalus:create_users(Config, [{eve, EveSpec2}]),
-    escalus:init_per_testcase(CN, [{evespec_reg, EveSpec}, {evespec_mim, EveSpec2} |Config]);
+    escalus:init_per_testcase(CN, init_user_eve(Config));
 init_per_testcase(CaseName, Config) ->
     escalus:init_per_testcase(CaseName, Config).
 
-end_per_testcase(test_pm_with_graceful_reconnection_to_different_server = CN, Config) ->
+init_user_eve(Config) ->
+    %% Register Eve in fed cluster
+    EveSpec = escalus_fresh:create_fresh_user(Config, eve),
+    MimPort = ct:get_config({hosts, mim, c2s_port}),
+    EveSpec2 = lists:keystore(port, 1, EveSpec, {port, MimPort}),
+    %% Register Eve in mim cluster
+    escalus:create_users(Config, [{eve, EveSpec2}]),
+    [{evespec_reg, EveSpec}, {evespec_mim, EveSpec2} | Config].
+
+end_per_testcase(CN, Config) when CN == test_pm_with_graceful_reconnection_to_different_server;
+                                  CN == test_pm_with_ungraceful_reconnection_to_different_server ->
```

Comment:
Here, `ungraceful` didn't have a special logic for the `end_per_testcase`, I tried that and had the same failure you have now 🤷‍♂